packages/ui-kit/src/components/mermaid.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import LoadingSpinner from '@commercetools-uikit/loading-spinner';
+
+// This is a client-side only implementation of mermaid, which is its primary development target.
+// https://www.gatsbyjs.com/docs/using-client-side-only-packages/
+// Server side rendering mermaid diagrams is technically possible
+// and some remark and gatsby plugins do it. But all are badly maintained
+// and the server side rendering use case is not covered in the mermaid documentation,
+// which makes it a hard to maintain architecture.
+// server side rendering also has the disadvantage that clickable elements cannot
+// work because this requires attaching event handlers.
+
+const MermaidLazy = React.lazy(() => import('./mermaid-client-side'));
```

Comment:
There are other choices for client-side only components in gatsbyJS, too.  React Lazy+Suspense is chosen because it does not incur extra libraries overhead, works and also provides lazy loaded bundle splitting for the big mermaidJS library.